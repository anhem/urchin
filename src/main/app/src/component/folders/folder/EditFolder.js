import React, {Component} from 'react'
import history from "../../../history";

class Folders extends Component {

    update = (e) => {
        let data = {
            field: e.target.name,
            value: e.target.value
        };

        this.props.callbacks.setFolder(data);
    };

    del = () => {
        this.props.callbacks.deleteFolder(this.props.folder.folderId);
    };

    back = () => {
        history.push('/folders');
    };

    render() {
        let folderName = this.props.folder.folderName;
        let folderPath = this.props.folder.folderPath;
        return (
            <div id="edit-folder">
                <h2>Folder</h2>
                <div>Name: {folderName}</div>
                <div>Path: {folderPath}</div>
                <button id="edit-folder__delete-btn" onClick={this.del}>Delete</button>
                <button id="edit-folder__back-btn" onClick={this.back}>Back</button>
            </div>
        )
    }
}

export default Folders