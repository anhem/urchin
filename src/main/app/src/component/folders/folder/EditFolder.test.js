import React from 'react';
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json';
import EditFolder from "./EditFolder";

describe('EditFolder', () => {

    let props = {
        folder: {},
        usersInFolder: [],
        availableUsers: []
    };

    it('match snapshot', () => {
        expect(toJson(shallow(<EditFolder {...props}/>))).toMatchSnapshot();
    });
});
