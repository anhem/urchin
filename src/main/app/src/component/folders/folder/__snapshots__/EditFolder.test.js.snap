// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditFolder match snapshot 1`] = `
<div
  id="edit-folder"
>
  <h2>
    Folder
  </h2>
  <div>
    Name: 
  </div>
  <div>
    Path: 
  </div>
  <button
    id="edit-folder__delete-btn"
    onClick={[Function]}
  >
    Delete
  </button>
  <button
    id="edit-folder__back-btn"
    onClick={[Function]}
  >
    Back
  </button>
</div>
`;
